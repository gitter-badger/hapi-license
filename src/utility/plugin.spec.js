import {
    createLogger,
    createGoodOptions,
    createSwaggerOptions,
    createDevPlugins,
    createLocalPlugins,
    createProductionPlugins
} from './plugin';

describe('server plugins', () => {
    it('should create a bunyan logger', () => {
        const bunyan = {
            createLogger: jest.fn(() => 'test')
        };

        expect(createLogger(bunyan)).toEqual('test');
        expect(bunyan.createLogger).toHaveBeenCalled();
        expect(bunyan.createLogger).toHaveBeenCalledTimes(1);

        const calls = bunyan.createLogger.mock.calls[0][0];
        expect(calls).toMatchSnapshot();
    });

    describe('plugin options', () => {
        it('should create good options', () => {
            expect(createGoodOptions('test')).toMatchSnapshot();
        });

        it('should create swagger options', () => {
            const serverInfo = {
                host: 'localhost:3000',
                scheme: 'http'
            };

            expect(createSwaggerOptions(serverInfo)).toMatchSnapshot();
        });
    });

    describe('plugin environments', () => {
        it('should create dev plugins', () => {
            const serverInfo = {
                host: 'localhost:3000',
                scheme: 'http'
            };

            expect(createDevPlugins(serverInfo)).toMatchSnapshot();
        });

        it('should create local plugins', () => {
            expect(createLocalPlugins()).toMatchSnapshot();
        });

        it('should create production plugins', () => {
            const logger = 'test';
            expect(createProductionPlugins(logger)).toMatchSnapshot();
        });
    });
});
