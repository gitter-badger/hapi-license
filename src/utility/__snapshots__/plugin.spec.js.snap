// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server plugins plugin environments should create dev plugins 1`] = `
Array [
  Object {
    "plugin": Object {
      "options": Object {
        "grouping": "tags",
        "host": "localhost:3000",
        "info": Object {
          "contact": Object {
            "name": "Josh Ferrell",
            "url": "http://google.com",
          },
          "description": "Licesning api that connects to auth0 for license information",
          "title": "undefined Documentation",
          "verson": "1.0.0",
        },
        "schemes": Array [
          "http",
        ],
        "securityDefinitions": Object {
          "manageLicense": Object {
            "in": "header",
            "name": "Authorization",
            "type": "apiKey",
          },
        },
      },
      "register": "hapi-swagger",
    },
  },
]
`;

exports[`server plugins plugin environments should create local plugins 1`] = `
Array [
  Object {
    "plugin": Object {
      "options": Object {
        "endpoint": "/tv",
      },
      "register": "tv",
    },
  },
]
`;

exports[`server plugins plugin environments should create production plugins 1`] = `
Array [
  Object {
    "plugin": "inert",
  },
  Object {
    "plugin": "vision",
  },
  Object {
    "plugin": "hapi-auth-jwt2",
  },
  Object {
    "plugin": Object {
      "options": Object {
        "includes": Object {
          "request": Array [
            "headers",
            "payload",
          ],
          "response": Array [
            "payload",
          ],
        },
        "ops": Object {
          "interval": 2160000,
        },
        "reporters": Object {
          "bunyan": Array [
            Object {
              "args": Array [
                Object {
                  "error": "*",
                  "log": "*",
                  "request": "*",
                  "response": "*",
                },
              ],
              "module": "good-squeeze",
              "name": "Squeeze",
            },
            Object {
              "args": Array [
                Object {
                  "Authorization": "censor",
                },
              ],
              "module": "white-out",
            },
            Object {
              "args": Array [
                Object {
                  "error": "*",
                  "log": "*",
                  "request": "*",
                  "response": "*",
                },
                Object {
                  "formatters": Object {
                    "err": [Function],
                  },
                  "logger": "test",
                },
              ],
              "module": "good-bunyan",
            },
          ],
        },
      },
      "register": "good",
    },
  },
]
`;

exports[`server plugins plugin options should create good options 1`] = `
Object {
  "includes": Object {
    "request": Array [
      "headers",
      "payload",
    ],
    "response": Array [
      "payload",
    ],
  },
  "ops": Object {
    "interval": 2160000,
  },
  "reporters": Object {
    "bunyan": Array [
      Object {
        "args": Array [
          Object {
            "error": "*",
            "log": "*",
            "request": "*",
            "response": "*",
          },
        ],
        "module": "good-squeeze",
        "name": "Squeeze",
      },
      Object {
        "args": Array [
          Object {
            "Authorization": "censor",
          },
        ],
        "module": "white-out",
      },
      Object {
        "args": Array [
          Object {
            "error": "*",
            "log": "*",
            "request": "*",
            "response": "*",
          },
          Object {
            "formatters": Object {
              "err": [Function],
            },
            "logger": "test",
          },
        ],
        "module": "good-bunyan",
      },
    ],
  },
}
`;

exports[`server plugins plugin options should create swagger options 1`] = `
Object {
  "grouping": "tags",
  "host": "localhost:3000",
  "info": Object {
    "contact": Object {
      "name": "Josh Ferrell",
      "url": "http://google.com",
    },
    "description": "Licesning api that connects to auth0 for license information",
    "title": "undefined Documentation",
    "verson": "1.0.0",
  },
  "schemes": Array [
    "http",
  ],
  "securityDefinitions": Object {
    "manageLicense": Object {
      "in": "header",
      "name": "Authorization",
      "type": "apiKey",
    },
  },
}
`;

exports[`server plugins should create a bunyan logger 1`] = `
Object {
  "level": "trace",
  "name": "licensing-api",
  "serialziers": Object {
    "err": [Function],
  },
}
`;
